controllers/ingress/controller.go

```diff
@@ -121,44 +121,35 @@ func WithWatchSettings(name types.NamespacedName) Option {
 
 // SetupWithManager sets up the controller with the Manager
 func (r *ingressController) SetupWithManager(mgr ctrl.Manager) error {
-	c, err := ctrl.NewControllerManagedBy(mgr).
+	r.Client = mgr.GetClient()
+	r.Scheme = mgr.GetScheme()
+
+	// cache frequently used object kinds
+	r.secretKind = generic.GVKForType[*corev1.Secret](r.Scheme).Kind
+	r.ingressKind = generic.GVKForType[*networkingv1.Ingress](r.Scheme).Kind
+	r.serviceKind = generic.GVKForType[*corev1.Service](r.Scheme).Kind
+	r.settingsKind = generic.GVKForType[*icsv1.Pomerium](r.Scheme).Kind
+	r.endpointsKind = generic.GVKForType[*corev1.Endpoints](r.Scheme).Kind
+	r.ingressClassKind = generic.GVKForType[*networkingv1.IngressClass](r.Scheme).Kind
+
```

Comment:
it doesn't look like we're using them beyond this function?

Reply:
also used in controllers/settings/controller.go. Figured this would be useful to have for the upcoming api changes, or if controller-runtime introduces a similar builtin, this would make it easy to swap out.